library/net.cpp

```diff
@@ -63,6 +65,94 @@ enum dnet_socket_state {
  * @ok will be set to 1 if given socket was successfully initialized (connected or made listened)
  */
 struct dnet_addr_socket {
+	dnet_addr_socket()
+	: s(-1),
+	 buffer(nullptr)
+	{}
+
+	~dnet_addr_socket() {
+		if (s >= 0)
```

Comment:
Please make a method `close` and use it here and there https://github.com/shaitan/elliptics/pull/6/files#diff-e26c73064bf13c8f39a0de2f72201ff7R588